packages/issuer-rest/lib/api.ts

```diff
@@ -97,37 +99,83 @@ export class RestAPI {
-      return response.send(await this._vcIssuer.issueCredentialFromIssueRequest(request as unknown as CredentialRequest))
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    this.express.get('/credential-request', async (request, _response) => {
+      this._vcIssuer.issueCredentialFromIssueRequest(request as unknown as CredentialRequest)
     })
   }
 
   private registerCredentialOfferEndpoint() {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
     this.express.get('/credential-offer/:pre_authorized_code', async (request, response) => {
       const preAuthorizedCode = request.params.pre_authorized_code
       const id = uuidv4()
       this.tokenToId.set(preAuthorizedCode, id)
       return response.send(createCredentialOfferDeeplink(preAuthorizedCode, this._vcIssuer._issuerMetadata))
     })
   }
+
+  private pushedAuthorizationEndpoint() {
+    const handleHttpStatus400 = async (req: Request, res: Response, next: NextFunction) => {
+      if (!req.body) {
+        return res.status(400).send({ error: 'invalid_request', error_description: 'Request body must be present' })
+      }
+      const required = ['client_id', 'code_challenge_method', 'code_challenge', 'redirect_uri']
+      const conditional = ['authorization_details', 'scope']
+      try {
+        validateRequestBody({ required, conditional, body: req.body })
+      } catch (e: unknown) {
+        return res.status(400).json({
+          error: 'invalid_request',
+          error_description: (e as Error).message,
+        })
+      }
+      return next()
+    }
+
+    this.express.post('/par', handleHttpStatus400, (req: Request, res: Response) => {
+      // FIXME Fake client for testing, it needs to come from a registered client
+      const client = {
+        scope: ['openid', 'test'],
+        redirectUris: ['http://localhost:8080/*', 'https://www.test.com/*', 'https://test.nl', 'http://*/chart', 'http:*'],
+      }
+
+      // For security reasons the redirect_uri from the request needs to be matched against the ones present in the registered client
+      const matched = client.redirectUris.filter((s: string) => new RegExp(s.replace('*', '.*')).test(req.body.redirect_uri))
+      if (!matched.length) {
+        return res.status(400).send({ error: 'invalid_request', error_description: 'redirect_uri is not valid for the given client' })
+      }
+
+      // The scopes from the request need to be matched against the ones present in the registered client
+      if (!req.body.scope.split(',').every((scope: string) => client.scope.includes(scope))) {
+        return res.status(400).send({ error: 'invalid_scope', error_description: 'scope is not valid for the given client' })
+      }
+
+      //TODO Implement authorization_details verification
+
+      const uuid = uuidv4()
+      const requestUri = `urn:ietf:params:oauth:request_uri:${uuid}`
```

Comment:
The base URL for sure would also need to be optionally provided by the developer. Meaning there should be a configuration option fo the base URI (so everything in front of the uuid). This also needs a TODO, as in the future you even might want to bind the request uri to a specific client for instance